### ISSUE TITLE
Strange behaviour using spy

### ISSUE THREAD

**Issue by matiux** [psalm/psalm-plugin-mockery#16]
I'm getting strange behaviour trying to use a spy.

Let me show an example:

```
/**
 * @psalm-suppress MissingConstructor
 */
class SpyTest extends TestCase
{
    /**
     * @test
     */
    public function spy(): void
    {
        $mockDepService = \Mockery::mock(DepService::class);
        $s1 = new Service($mockDepService);

        $spyDepService = \Mockery::spy(DepService::class);
        $s2 = new Service($spyDepService);
    }
}

class DepService
{
}

class Service
{
    public function __construct(private DepService $depService)
    {
    }
}
```

By running Psalm on this code I get an error on the spy but not on the mock.

```
ERROR: InvalidArgument - tests/SpyTest.php:23:27 - Argument 1 of Tests\Service::__construct expects Tests\DepService, Mockery\MockInterface provided (see https://psalm.dev/004)
        $s2 = new Service($spyDepService);
```

Am I doing something wrong?

**Comment by weirdan** [comment]
Fixed in #22

**Comment by weirdan** [comment]
Fix released in [0.9.0](https://github.com/psalm/psalm-plugin-mockery/releases/tag/0.9.0)

**Comment by matiux** [comment]
Thanks @weirdan
